result/src/main/kotlin/com/github/kittinunf/result/Result.kt

```diff
@@ -112,6 +112,13 @@ inline fun <V : Any?, reified E : Exception> List<Result<V, E>>.lift(): Result<L
     }
 }
 
+inline fun <V, E : Exception> Result<V, E>.unwrap(failure: (E) -> Nothing): V =
```

Comment:
I am having hard time understand why this is better than the current API that we already have?

Reply:
How can I do:
```kotlin
val myValue = result.unwrap { throw RuntimeException(it) }
```

Reply:
@kittinunf